grails-app/controllers/org/pih/warehouse/core/StockMovementItemsRequestCommand.groovy

```diff
@@ -0,0 +1,19 @@
+package org.pih.warehouse.core
+
+import grails.validation.Validateable
+import org.pih.warehouse.api.PaginationCommand
+
+class StockMovementItemsRequestCommand extends PaginationCommand implements Validateable {
```

Comment:
Overall I think this looks good to me, but I am slightly afraid, that `StockMovementItemsRequestCommand` will be confused with the Requisition domain

Reply:
some potential names to replace the `StockMovementItemsRequestCommand`
- StockMovementItemsControllerCommand
- StockMovementItemsQueryCommand
- StockMovementItemsParamsCommand
WDYT?

Reply:
> [!IMPORTANT]
> **TL;DR** I'm going to give this a bit more thought, but I'm cool with StockMovementItemQueryCommand. Note the pluralization. 

TIL: Holy crap, look at this shit ^^^^^^ (had no idea we could do this)

I think there are a few conventions that we could consider here.

When all else fails and we don't have a good idea of a name based on the intent of the command class, then we naming it the same as the method is an ok fallback. I really don't mind this at all, to be honest. 

* GetStockMovementItemsCommand

The problem is when the same command is used across methods (like we do here with the details action) we could start thinking of a better name, one that captures the essence of what the command class is intended to be used for. 

In this case it probably makes sense to call it something more expressive.
* StockMovementItemQueryCommand (recommended)
* StockMovementItemSearchCommand (meh)
* StockMovementItemFiltersCommand (we've used this in the past, but it might be too narrow)

Note: I think "Request" and "Params" are telling us what the object is, not what it is for. That might come in handy for a base class or when the intention is multi-faceted (we want to query or create with the same command).

We could also just start with something basic like the name of the name of the underlying class and then refactor as needed.

* StockMovementItemCommand

This is generic enough to carry individual fields about a stock movement item for creation as well as properties for filtering/querying.

Reply:
i've added a convention document for this, so we can update that if/when we decide on something
https://pihemr.atlassian.net/wiki/spaces/OB/pages/3075997703/Command+Class